contracts/Nexus.sol

```diff
@@ -313,7 +313,7 @@ contract Nexus is INexus, BaseAccount, ExecutionHelper, ModuleManager, UUPSUpgra
     /// as Biconomy v2 Account (proxy) reads implementation from the slot that is defined by its address
     /// @param newImplementation The address of the new contract implementation.
     /// @param data The calldata to be sent to the new implementation.
-    function upgradeToAndCall(address newImplementation, bytes calldata data) public payable virtual override onlyEntryPointOrSelf withHook {
```

Comment:
why is this redundant?

Reply:
it's already on `_authorizeUpgrade`

Reply:
`    function _authorizeUpgrade(address newImplementation) internal virtual override(UUPSUpgradeable) onlyEntryPointOrSelf { }`

Reply:
remember discussing this with Konrad/Jon. 
I think we can remove from _authorizeUpgrade and bring back in public method

Reply:
I decided to do it this way to comply with Solady's recommendation

```

    /// @dev Please override this function to check if `msg.sender` is authorized
    /// to upgrade the proxy to `newImplementation`, reverting if not.
    /// ```
    ///     function _authorizeUpgrade(address) internal override onlyOwner {}
    /// ```
    function _authorizeUpgrade(address newImplementation) internal virtual;

```

Reply:
hmm had done this way before. but to think of it internal methods are not touched so naturally makes sense to apply on public main method. 
if needed, get more opinions on this from our team and rhinestone. 
also remember this code is already audited and will cause changed to be added in requested new review.

Reply:
ok reverting this cause dont want to confuse spearbit.